lib/integrations/CalDav/components/AddCalDavIntegration.tsx

```diff
@@ -11,8 +24,106 @@ export type AddCalDavIntegrationRequest = {
   password: string;
 };
 
+export function AddCalDavIntegrationModal(props: DialogProps) {
+  const form = useForm({
+    defaultValues: {
+      url: "",
+      username: "",
+      password: "",
+    },
+  });
+  const [errorMessage, setErrorMessage] = useState("");
+  return (
+    <Dialog {...props}>
+      <DialogContent>
+        <DialogHeader
+          title="Connect to CalDav Server"
+          subtitle="Your credentials will be stored and encrypted."
+        />
+
+        <Form
```

Comment:
Idea for the onSubmit -> handleSubmit can be called automatically in Form so you can just do your handler without bothering to manually call the wrapper.

Reply:
I'd like that as well but I was fighting with types for too long - can do as follow-up